boa/src/object/jsarray.rs

```diff
@@ -0,0 +1,344 @@
+use std::ops::Deref;
+
+use crate::{
+    builtins::Array,
+    gc::{Finalize, Trace},
+    object::JsObject,
+    Context, JsResult, JsString, JsValue,
+};
+
+// TODO: put in a better place.
+pub trait JsObjectType: Into<JsValue> + Into<JsObject> + Deref<Target = JsObject> {}
+
+/// JavaScript `Array` rust object.
+#[derive(Debug, Clone, Trace, Finalize)]
+pub struct JsArray {
+    inner: JsObject,
+}
+
+impl JsArray {
+    #[inline]
+    pub fn new(array: &[JsValue], context: &mut Context) -> Self {
```

Comment:
Maybe this should have the same API as `create_array_from_list`. That way users can avoid cloning whole collections if they want to

https://github.com/boa-dev/boa/blob/b34d581db50bdef70e6598f96bcf04c19b618341/boa/src/builtins/array/mod.rs#L249-L251